.buildkite/pipeline.yml

```diff
@@ -240,6 +243,23 @@ steps:
       architecture: CPU
     depends_on: "init_cpu"
 
+#####
+##### Distributed/MPI
+#####
+
+  - label: "🐉 cpu distributed tests"
+    env:
+      JULIA_DEPOT_PATH: "$TARTARUS_HOME/.julia-$BUILDKITE_BUILD_NUMBER"
+      TEST_GROUP: "distributed"
+      CUDA_VISIBLE_DEVICES: "-1"
+    commands:
+      - "PATH=$PATH:$TARTARUS_HOME/julia-$JULIA_VERSION/bin"  # Need julia binary in $PATH for mpiexecjl to work.
+      - "$TARTARUS_HOME/.julia-$BUILDKITE_BUILD_NUMBER/bin/mpiexecjl -np 4 $TARTARUS_HOME/julia-$JULIA_VERSION/bin/julia -O0 --color=yes --project -e 'using Pkg; Pkg.test()'"
```

Comment:
@ali-ramadhan one more annoying comment! It would be good to test on 1, 2, 3 and 4 ranks.

Reply:
Yes!

I asked about how to generalize the tests to multiple ranks on the CliMA Slack (#clima-helpdesk channel?) and the recommended approach seems to be to launch multiple `mpiexec -np N` processes with different values of `N` (instead of launching one `mpiexec -np N_max` process and using lots of `if my_rank < m` statements, which wouldn't work if you have barriers in specific spots...).

To me it seems like we can't exactly reuse tests for 1, 2, 3, 4, etc. ranks so maybe there will be some necessary boilerplate or copy-pasting, which I'm fine with, but I felt I might not have enough time to add all the extra tests right away.

Would testing non-equal domain decompositions and multiple ranks be a good goal for the distributed shallow water model PR (#1430) or should it be part of #590?

Reply:
In theory if #590 works that is a good foundation. But #1430 would be nice to have too!